file_converter/Program.cs

```diff
@@ -2,94 +2,73 @@
-                        case "json":
-                            inFtype = new file_json(args[0]);
-                            break;
-                        case "xml":
-                            inFtype = new file_xml(args[0]);
-                            break;
-                        default:
-                            break;
-                    }
-                    try
-                    {
-                        mydata = inFtype.parser();
-                    }
-                    catch
-                    {
-                        Console.WriteLine("Can`t parse {0}", args[0]);
-                    }
-                    
+            else
+            {
+                Data myData = new Data();
 
-                    //export
-                    switch (arg_o[arg_o.Length - 1])
-                    {
-                        case "txt":
-                            outFtype = new file_txt(args[1]);
-                            break;
-                        case "csv":
-                            outFtype = new file_csv(args[1]);
-                            break;
-                        case "json":
-                            outFtype = new file_json(args[1]);
-                            break;
-                        case "xml":
-                            outFtype = new file_xml(args[1]);
-                            break;
-                        default:
-                            break;
-                    }
-                    try
-                    {
-                        outFtype.exporter(mydata);
-                    }
-                    catch
-                    {
-                        if (arg_o[arg_o.Length - 1] == "xml")
-                            Console.WriteLine("Can`t export...\nMaybe file {0} has no root \nOR \n{1} has bad structure", args[1], args[0]);
-                        else
-                            Console.WriteLine("Can`t export...\nMaybe file {0} has bad structure", args[0]);
-                    }
+                IFileWrapper inFtype = new fileTxt();
+                IFileWrapper outFtype = new fileTxt();
 
-                    Console.WriteLine("Done!");
-                    Console.ReadLine();
+                inFtype = BuildFileWrapper(args[0]);
+                try
+                {
+                    myData = inFtype.Parse();
                 }
-                else
+                catch
                 {
-                    Console.WriteLine("Can`t convert such type of file :C");
-                    Console.ReadLine();
+                    Console.WriteLine("Can`t parse {0}", args[0]);
                 }
+
+                outFtype = BuildFileWrapper(args[1]);
+                //try
+                //{
+                outFtype.Export(myData);
+                //}
+                //catch
+                //{
+                //    Console.WriteLine("Can`t export...\nMaybe file {0} has bad structure", args[0]);
+                //}
```

Comment:
It can be simplified:
```csharp
try
{
   Data myData = inFtype.Parse();
   outFtype.Export(myData);
}
catch
{
    Console.Error.WriteLine("Something went wrong during file conversion");
}
```